python/pyarrow/includes/libarrow.pxd

```diff
@@ -1784,11 +1788,26 @@ cdef extern from "arrow/json/reader.h" namespace "arrow::json" nogil:
         CResult[shared_ptr[CTable]] Read()
 
 
+cdef extern from "arrow/util/thread_pool.h" namespace "arrow::internal" nogil:
```

Comment:
They have different namespaces by the way, the `GetCpuThreadPoolCapacity` is `arrow::` while the `CExecutor` is `arrow:internal::`

Reply:
> They have different namespaces by the way, the `GetCpuThreadPoolCapacity` is `arrow::` while the `CExecutor` is `arrow:internal::`

That's more a C++ question, but if we use this here, shouldn't that ideally be moved in the arrow namespace instead of the internal?(cc @pitrou)